daq/faucet_state_collector.py

```diff
@@ -308,21 +370,12 @@ def get_access_switch(self, mac):
         learned_switches = self.learned_macs.get(mac, {}).get(KEY_MAC_LEARNING_SWITCH)
 
         for switch, port_map in learned_switches.items():
-            access_switch_port[switch] = port_map[KEY_MAC_LEARNING_PORT]
-
+            port = port_map[KEY_MAC_LEARNING_PORT]
+            port_attr = self.get_port_attributes(switch, port)
+            if port_attr['type'] == 'access':
+                access_switch_port[switch] = port
```

Comment:
Can we directly return the switch and port here?

Reply:
Fair point